packages/features/users/components/UserTable/UserListTable.tsx

```diff
@@ -0,0 +1,353 @@
+import type { ColumnDef } from "@tanstack/react-table";
+import { Plus, StopCircle, Users } from "lucide-react";
+import { useSession } from "next-auth/react";
+import { useMemo, useRef, useCallback, useEffect, useReducer } from "react";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { MembershipRole } from "@calcom/prisma/enums";
+import { trpc } from "@calcom/trpc";
+import { Avatar, Badge, Button, ConfirmationDialogContent, DataTable, Dialog, showToast } from "@calcom/ui";
+
+import { ChangeUserRoleModal } from "./ChangeUserRoleModal";
+import { ImpersonationMemberModal } from "./ImpersonationMemberModal";
+import { InviteMemberModal } from "./InviteMemberModal";
+import { TableActions } from "./UserTableActions";
+
+export interface User {
+  id: number;
+  username: string | null;
+  email: string;
+  timeZone: string;
+  role: MembershipRole;
+  accepted: boolean;
+  disableImpersonation: boolean;
+  teams: {
+    id: number;
+    name: string;
+    slug: string | null;
+  }[];
+}
+
+type Payload = {
+  showModal: boolean;
+  user?: User;
+};
+
+export type State = {
+  changeMemberRole: Payload;
+  deleteMember: Payload;
+  impersonateMember: Payload;
+  inviteMember: Payload;
+};
+
+export type Action =
+  | {
+      type: "SET_CHANGE_MEMBER_ROLE_ID" | "SET_DELETE_ID" | "SET_IMPERSONATE_ID" | "INVITE_MEMBER";
+      payload: Payload;
+    }
+  | {
+      type: "CLOSE_MODAL";
+    };
+
+const initialState: State = {
+  changeMemberRole: {
+    showModal: false,
+  },
+  deleteMember: {
+    showModal: false,
+  },
+  impersonateMember: {
+    showModal: false,
+  },
+  inviteMember: {
+    showModal: false,
+  },
+};
```

Comment:
Instead of a bunch of state i changed this to reducer to make this easier to work with.